docs/content/v0.20/architecture.md

```diff
@@ -444,3 +484,25 @@ of tracked objects in the namespace associated with the WEC cluster.
 A `WorkStatus` object contains status for exactly one object, so that 
 status updates for one object do not require updates of a whole bundle. 
 
+
+### Transport Controller - PLANNED TO BE ADDED
+
+The transport controller is pluggable and allows the option to plug different
+implementations of the transport interface. In order to implement transport plugin, the plugin should supply the following:
+- upon registration of a new WEC, plugin should create a namespace per WEC in ITS (mailbox namespace per WEC);
+- plugin must be able to wrap multiple objects into a single wrapped object;
+- have an agent that runs on the WEC and watches the wrapped object in the corresponding namespace in the central hub and is able to unwrap it and apply the objects to the WEC;
+- have inventory representation for the clusters.
+
+The above list is required in order to comply with [<u>SIG Multi-Cluster Work API</u>](https://multicluster.sigs.k8s.io/concepts/work-api/).
+
+KubeStellar currently have one transport plugin implementation that is based on [Open Cluster Management](https://open-cluster-management.io) which is also a CNCF Sandbox project. we expect to have more transport plugin options in the future.
+
+The following section describes how transport controller works, while the described behavior remains the same no matter which transport plugin is selected. The high level flow for the transport controller is describted in Figure 5.
+
+<figure>
+  <img src="./images/transport-controller.png"  alt="Transport Controller">
+  <figcaption align="center">Figure 5 - Transport Controller</figcaption>
+</figure>
+
+The transport controller watches for `Binding` objects on the WDS. `Binding` objects are mapped 1:1 to `BindingPolicy` and contain the list of concrete objects that were selected for distribution by the policy, and the concrete list of destination that were selected by the policy. upon a new `Binding` event (add/update), the transport controller gets from the WDS api-server the objects listed in the `Binding` workload section. it then wraps all objects into a single wrapped object and puts a copy of the wrapped object in every matching cluster mailbox namespace in the ITS. once the wrapped object is in the mailbox namespace of a cluster on the ITS, it's the agent responsibility to pull the wrapped object from there and apply/update/remove the workload objects. 
```

Comment:
"into a single wrapped object" is too specific, right? I thought we agreed that wrapping them individually would be an option.

Reply:
fixed

Reply:
My original comment here was based on a misunderstanding of what you were trying to document. If I understand your intent correctly now then no revision was needed here, and the revision that was done made the text more ambiguous. This is not technically incorrect.

Reply:
I reverted back to previous version. I also thought about it later and I agree we don't need this change.
this is in the section `Transport Controller - Planned to be added`. This does **NOT** represent current main. it's the planned changes (as you requested, we added the aspirational part).